### ISSUE TITLE
Port twisted.internet.test.test_threads to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6018]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6018|
|Type|enhancement|
|Created|2012-09-24 18:10:02Z|
|Branch|https://github.com/twisted/twisted/tree/testthreads-py3-6018|

`twisted.internet.test.test_threads` should run on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__6018 6018
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_testthreads_py3_6018 branches/testthreads-py3-6018
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1348510202000000 1348510202000000
changetime__1348858717000000 1348858717000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35809]) Branching to 'testthreads-py3-6018'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

Started build: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/testthreads-py3-6018

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Failing tests

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @exarkun|
|-|-|

The failing tests were due to, and fixed in, [#6005](https://github.com/twisted/twisted/issues/6005); they won't be present when I merge to trunk.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @itamarst|
|-|-|

Okay, great.  Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst set status to `closed`|
|-|-|

(In [35836]) Merge testthreads-py3-6018

Author: itamar
Review: exarkun
Fixes: [#6018](https://github.com/twisted/twisted/issues/6018)

twisted.internet.test.test_threads now runs on Python 3.
